src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/Saml2SecurityToken.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.ObjectModel;
+using CoreWCF.Security;
+using MSIdentitySAML2 = Microsoft.IdentityModel.Tokens.Saml2;
+
+namespace CoreWCF.IdentityModel.Tokens
+{
+    /// <summary>
+    /// A security token backed by a SAML2 assertion.
+    /// </summary>
+    public class Saml2SecurityToken : SecurityToken
+    {
+        protected Saml2SecurityToken()
+        {
+        }
+
+        public Saml2SecurityToken(MSIdentitySAML2.Saml2SecurityToken saml2SecurityToken)
+        {
+            WrappedSaml2SecurityToken = saml2SecurityToken;
+            Assertion = WrappedSaml2SecurityToken.Assertion;
+
+        }
+
+        public Saml2SecurityToken(MSIdentitySAML2.Saml2Assertion assertion)
+        {
+            Initialize(assertion);
+        }
+
+        protected void Initialize(MSIdentitySAML2.Saml2Assertion assertion)
+        {
+            Assertion = assertion ?? throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(assertion));
+        }
+
+        internal string AssertionXML { get; set; }
+
+        public override string Id
+        {
+            get { return Assertion.Id.Value; }
+        }
+
+        //public override ReadOnlyCollection<SecurityKey> SecurityKeys
+        //{
```

Comment:
Can you remove this commented out version of the property? It exists throwing at the bottom of this file.